### ISSUE TITLE
Surface pro 6 powersave not working

### ISSUE THREAD

**Issue by ttoommxx** [linux-surface/linux-surface#1285]
Running `surface profile set low-power` with sudo privileges returns that it has been set correctly, however it hasn't and the profile stays balanced. The guide says that if LE bluetooth devices are paired this behaviour is to be expected, but I have no bluetooth device associated at all and bluetooth is actually off. Any fix that I am missing? Or shall we modify the feature matrix and write that powersave profile is simply broken?

**Comment by NiklasVoigt** [comment]
Same here with Surface Pro 6 and manjaro

**Comment by qzed** [comment]
So there are two possibilities that I can see happening: Either something (e.g. Gnome or its power daemon) changes the mode back, or the EC ignores the command (and doesn't respond with an error code). The latter might be influenced by the low-power mode only being available when the AC is disconnected on Windows, so you could maybe check if that has some influence on it. You can also check dmesg and journalctl to maybe figure out if there's an error in setting the mode or if something changes it back later.

In short, this command basically just instructs the EC to change the power mode. So if the EC ignores setting the low-power mode, then there's really nothing we can do to fix it. Maybe we could check in the driver whether the mode has been applied correctly and the value changed by reading it back (throwing an error if it hasn't changed), but that's about all we could do.

**Comment by qzed** [comment]
Hard to tell for me if Gnome blocks something there. I'm also not sure how to best investigate this. Ideally, you'd need to check who accesses `/sys/firmware/acpi/platform_profile`.

Does the message also appear when you run `surface profile set low-power`, i.e. don't use Gnome to change it?

**Comment by ttoommxx** [comment]
It does not! But also still does not set the correct profile and stays on balanced.. thank you for your help btw :)

**Comment by qzed** [comment]
Okay, so I checked and you could try to monitor file reads and writes via the following command:
```
inotifywait -m /sys/firmware/acpi/platform_profile
```

Run that in a terminal and then change the profile. This should normally only show a single write access (but can show multiple reads). If there's more than one write access, I assume it's something in user-space resetting the mode. If not, then I guess it's the EC not accepting it.

**Comment by ttoommxx** [comment]
So, triggering powersave from the Gnome setting panel shows the following:
`Setting up watches.
Watches established.
/sys/firmware/acpi/platform_profile OPEN 
/sys/firmware/acpi/platform_profile MODIFY 
/sys/firmware/acpi/platform_profile MODIFY 
/sys/firmware/acpi/platform_profile CLOSE_WRITE,CLOSE 
/sys/firmware/acpi/platform_profile MODIFY 
/sys/firmware/acpi/platform_profile OPEN 
/sys/firmware/acpi/platform_profile ACCESS 
/sys/firmware/acpi/platform_profile CLOSE_NOWRITE,CLOSE 
/sys/firmware/acpi/platform_profile OPEN 
/sys/firmware/acpi/platform_profile ACCESS 
/sys/firmware/acpi/platform_profile CLOSE_NOWRITE,CLOSE 
/sys/firmware/acpi/platform_profile OPEN 
/sys/firmware/acpi/platform_profile ACCESS 
/sys/firmware/acpi/platform_profile CLOSE_NOWRITE,CLOSE 
/sys/firmware/acpi/platform_profile OPEN 
/sys/firmware/acpi/platform_profile ACCESS 
/sys/firmware/acpi/platform_profile CLOSE_NOWRITE,CLOSE 
/sys/firmware/acpi/platform_profile OPEN 
/sys/firmware/acpi/platform_profile ACCESS 
/sys/firmware/acpi/platform_profile CLOSE_NOWRITE,CLOSE 
`
and triggering low-power from the surface control cli tool triggers the following:
`Setting up watches.
Watches established.
/sys/firmware/acpi/platform_profile OPEN 
/sys/firmware/acpi/platform_profile ACCESS 
/sys/firmware/acpi/platform_profile CLOSE_NOWRITE,CLOSE 
/sys/firmware/acpi/platform_profile OPEN 
/sys/firmware/acpi/platform_profile ACCESS 
/sys/firmware/acpi/platform_profile CLOSE_NOWRITE,CLOSE 
/sys/firmware/acpi/platform_profile OPEN 
/sys/firmware/acpi/platform_profile MODIFY 
/sys/firmware/acpi/platform_profile CLOSE_WRITE,CLOSE 
/sys/firmware/acpi/platform_profile OPEN 
/sys/firmware/acpi/platform_profile ACCESS 
/sys/firmware/acpi/platform_profile CLOSE_NOWRITE,CLOSE 
/sys/firmware/acpi/platform_profile OPEN 
/sys/firmware/acpi/platform_profile ACCESS 
/sys/firmware/acpi/platform_profile CLOSE_NOWRITE,CLOSE 
`

**Comment by qzed** [comment]
Also this seems to be a duplicate of #811. So let's continue this there.
